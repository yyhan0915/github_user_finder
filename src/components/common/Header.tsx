import * as React from 'react';
import { Link } from 'react-router-dom';
import styled, { StyledComponent } from 'styled-components';

import { Button } from '../';

import Responsive, { ResponsiveProps } from './Responsive';

const HeaderBlock = styled.div`
  position: fixed;
  width: 100%;
  background: white;
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.08);
`;

const Wrapper: StyledComponent<React.FC<ResponsiveProps>, any> = styled(
  Responsive,
)`
  height: 4rem;
  display: flex;
  align-items: center;
  justify-content: space-between;
  .logo {
    font-size: 1.125rem;
    font-weight: 800;
    letter-spacing: 1px;
  }
  .right {
    display: flex;
    align-items: center;
  }
`;

const Spacer = styled.div`
  height: 4rem;
`;

interface HeaderProps {
  // 나중에 채울 것
}
export const Header: React.FC<HeaderProps> = () => {
  return (
    <>
      <HeaderBlock>
        <Wrapper>
          <Link to="/" className="logo">
            GITHUB USER FINDER
          </Link>
          <div className="right">
            <Button cyan={1}>UserSample</Button>
          </div>
        </Wrapper>
      </HeaderBlock>
      <Spacer />
    </>
  );
};
