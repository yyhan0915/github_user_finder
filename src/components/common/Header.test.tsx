import * as React from 'react';
import { BrowserRouter } from 'react-router-dom';
import * as renderer from 'react-test-renderer';

import { Header } from './Header';

test('has a valid snapshot', () => {
    const component = renderer.create(
        <BrowserRouter>
            <Header />
        </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
