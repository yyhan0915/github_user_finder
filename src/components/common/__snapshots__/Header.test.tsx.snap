// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has a valid snapshot 1`] = `
Array [
  <div
    className="sc-AxmLO kyACbO"
  >
    <div
      className="sc-AxiKw kAmtBG sc-fzozJi dhrCMu"
    >
      <a
        className="logo"
        href="/"
        onClick={[Function]}
      >
        <h2>
          GITHUB USER FINDER
        </h2>
      </a>
      <div
        className="right"
      >
        <button
          className="sc-AxjAm UyhcK"
        >
          <a
            href="http://github.com"
          >
            Go to GITHUB
          </a>
        </button>
      </div>
    </div>
  </div>,
  <div
    className="sc-fzoLsD btLzaa"
  />,
]
`;
